Bugfix/update off chain replicate statuses (#158)
iExecBlockchainComputing/documentation: for-developers/advanced/task-feedback.md

| Replicate status | Description |
| --- | --- |
| `CREATED` | A new _replicate_ is assigned to a worker just after it asked for more work |
| `STARTING` | The worker starts preflight checks to confirm it can work on this _replicate_ |
| `STARTED` | The worker confirms it is going to work on this _replicate_ |

> | `START_FAILED` | The worker confirms it is going to work on this _replicate_ |

Reviewer: ```suggestion
| `START_FAILED` | The preflight checks have failed. The worker will NOT work on this _replicate_ |
```

| `COMPUTING` | The worker is computing the _task_ |
| `COMPUTED` | The computation is completed |
| `COMPUTE_FAILED` | The computation failed |
| `CONTRIBUTING` | The worker sent the "contribute(..)" transaction (result digest) on chain |
| `CONTRIBUTE_FAILED` | The contribute transaction failed |
| `CONTRIBUTED` | The worker has contributed on chain |
| `REVEALING` | The worker sent the "reveal(..)" transaction (proof that he is the owner of the result digest) |
| `REVEALED` | The worker has revealed the proof on chain |
| `REVEAL_FAILED` | The reveal transaction failed |
| `RESULT_UPLOAD_REQUESTED` | The worker has been requested to upload the result to a remote filesystem |
| `RESULT_UPLOAD_REQUEST_FAILED` | The worker did not accept to be requested to upload the result |
| `RESULT_UPLOADING` | The worker is uploading the result |
| `RESULT_UPLOAD_FAILED` | The upload of the result failed |

> | `RESULT_UPLOADED` | The result is uploaded to IPFS (over the _iExec Result Proxy_) |

Reviewer: What about Dropbox?

Author: Sure, see bef4ae3 for a more accurate description

Reviewer: Thanks for the proposal. I would prefer both sentences to be merged. Something as follows:
> The result has been uploaded to IPFS (over the _iExec Result Proxy_) or to Dropbox, depending on the task configuration

What's your opinion on that version?

Author: Let me thinks about this, I would like to be a bit more precise on where the result can be published (IPFS for standard and TEE, Dropbox for TEE only)

Author: What do you think of 26a5560 ?

| `COMPLETING` | The _task_ is finalized, the worker will purge data related to its _replicate_ |
| `COMPLETED` | The whole _task_ is completed meaning the _task_ is finalized. The worker has been rewarded if it is part of the consensus |

> | `COMPLETE_FAILED` | The worker failed to clean the local _replicate_ resources after the _task_ is finnalized |

Reviewer 2: ```suggestion
| `COMPLETE_FAILED` | The worker failed to clean the local _replicate_ resources after the _task_ is finalized |
```

| `FAILED` | The worker failed to participate to the _task_ |
| `ABORTED` | The scheduler asked the worker to stop working on this _replicate_ while the latter was still working on it |
| `RECOVERING` | The worker has been stopped, it is starting back from where it stopped |
| `WORKER_LOST` | The worker didn't ping the iexec-core scheduler for a while. It is considered as out for this _task_ |

The transitions between those states are as follow:

```mermaid
flowchart

subgraph Contribute stage
    CREATED --> STARTING
    STARTING --> STARTED
    STARTED --> APP_DOWNLOADING
    APP_DOWNLOADING --> APP_DOWNLOADED
    APP_DOWNLOADED --> DATA_DOWNLOADING
    DATA_DOWNLOADING --> DATA_DOWNLOADED
    DATA_DOWNLOADED --> COMPUTING
    COMPUTING --> COMPUTED
    COMPUTED --> CONTRIBUTING
    CONTRIBUTING --> CONTRIBUTED

    STARTING --> START_FAILED
    APP_DOWNLOADING --> APP_DOWNLOAD_FAILED
    DATA_DOWNLOADING --> DATA_DOWNLOAD_FAILED
    COMPUTING --> COMPUTE_FAILED
    CONTRIBUTING --> CONTRIBUTE_FAILED
end

subgraph Reveal stage
    CONTRIBUTED --> REVEALING
    REVEALING --> REVEALED
    REVEALING --> REVEAL_FAILED
end

subgraph Result upload stage

>     REVEALED -- Only one contributing worker will upload results --> RESULT_UPLOAD_REQUESTED

Reviewer 2: ```suggestion
    REVEALED -- Only one contributing worker will upload the result --> RESULT_UPLOAD_REQUESTED
```

    RESULT_UPLOAD_REQUESTED --> RESULT_UPLOADING
    RESULT_UPLOADING --> RESULT_UPLOADED
    RESULT_UPLOADING --> RESULT_UPLOAD_FAILED
end

RESULT_UPLOADED --> COMPLETING

> REVEALED -- Most workers do not upload results --> COMPLETING

Reviewer 2: ```suggestion
REVEALED -- Most workers do not upload their result --> COMPLETING
```

COMPLETING --> COMPLETED

COMPLETING --> COMPLETE_FAILED

START_FAILED --> ABORTED
APP_DOWNLOAD_FAILED --> ABORTED
DATA_DOWNLOAD_FAILED --> ABORTED
COMPUTE_FAILED --> ABORTED
CONTRIBUTE_FAILED --> ABORTED
REVEAL_FAILED --> ABORTED
RESULT_UPLOAD_FAILED --> ABORTED
COMPLETE_FAILED --> ABORTED
```

## Off-chain replicates failure causes

> ### Failures from Scheduler

Reviewer 2: ```suggestion
### Failures detected by the Scheduler
```

| Replicate failure cause | Description |
| --- | --- |
| `REVEAL_TIMEOUT` | The worker took too long to reveal its proof (more than 2 periods after the consensus) |

> ### Failures from Worker

Reviewer 2: ```suggestion
### Failures notified by the Worker
```

A _replicate_ can fail with the following causes:

{% hint style="info" %}

In the following table, the task type column behaves as follow:

- `STD` designates failures only triggered by _standard tasks_.
- `TEE` designates failures only triggered by _TEE tasks_.
- Failures without an explicit type can appear for _standard_ and _TEE_ tasks.

{% endhint %}

> | Replicate failure cause | Replicate status | Task type | Description |

Reviewer 2: I suggest to split this table in three tables as your intro suggests:
1. `STD`
2. `TEE`
3.  `STD` & `TEE`

Removing "Task type" would allow more readability in Gitbook (+ avoids "infinite" scrolling in 1 table)

Author: I tried this at first and decided to merge everything at the end because it felt weird to me.
I will try to show an example with split tables and we will decide with everyone the final format.

Author: split done in 1aa1877

| --- | --- | --- | --- |
| `CHAIN_UNREACHABLE` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The task model could not be fetched from the blockchain |
| `STAKE_TOO_LOW` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | Worker deposit is too low |
| `TASK_NOT_ACTIVE` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | On-chain task status is not `ACTIVE` |
| `CONTRIBUTION_TIMEOUT` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | Contribution deadline has already been reached |
| `CONTRIBUTION_ALREADY_SET` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The worker has already contributed |

> | `WORKERPOOL_AUTHORIZATION_NOT_FOUND` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The authorization to contribute to the task is missing |

Reviewer 2: ```suggestion
| `WORKERPOOL_AUTHORIZATION_NOT_FOUND` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The workerpool authorization to contribute to the task is missing |
```

> | `TASK_DESCRIPTION_INVALID` | `STARTING` | `STD` | The task description contains inconsistencies and cannot be executed |

Reviewer 2: ```suggestion
| `TASK_DESCRIPTION_INVALID` | `STARTING` | `STD` | The task cannot be executed due to inconsistencies in its description |
```

> | `TEE_NOT_SUPPORTED` | `STARTING` | `TEE` | The current worker cannot execute TEE tasks |

Reviewer 2: ```suggestion
| `TEE_NOT_SUPPORTED` | `STARTING` | `TEE` | The current worker is not well configured to execute TEE tasks (or not compatible at all) |
```

| `UNKNOWN_SMS` | `STARTING` | `TEE` | SMS URL could not be resolved for this task |
| `GET_TEE_SERVICES_CONFIGURATION_FAILED` | `STARTING` | `TEE` | Failed to fetch TEE task configuration properties from SMS |

> | `APP_IMAGE_DOWNLOAD_FAILED` | `APP_DOWNLOADING` | | The download of the application `OCI` image failed |

Reviewer 2: ```suggestion
| `APP_IMAGE_DOWNLOAD_FAILED` | `APP_DOWNLOADING` | | The download of the application image failed |
```

Author: is there a problem with mentioning we are working with `OCI` images here ?

Author: References to `OCI` standard removed in fd63ee7

| `TEE_SESSION_GENERATION_SECURE_SESSION_STORAGE_CALL_FAILED` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_SECURE_SESSION_GENERATION_FAILED` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_SECURE_SESSION_NO_TEE_PROVIDER` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_SECURE_SESSION_UNKNOWN_TEE_PROVIDER` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_GET_TASK_DESCRIPTION_FAILED` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_NO_SESSION_REQUEST` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_NO_TASK_DESCRIPTION` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_GET_SESSION_FAILED` | `COMPUTING` | `TEE` | |
| `TEE_SESSION_GENERATION_UNKNOWN_ISSUE` | `COMPUTING` | `TEE` | |
| `PRE_COMPUTE_INVALID_ENCLAVE_CONFIGURATION` | `COMPUTING` | `TEE` | Application enclave configuration is invalid |
| `PRE_COMPUTE_INVALID_ENCLAVE_HEAP_CONFIGURATION` | `COMPUTING` | `TEE` | Application requested HEAP size is bigger than maximum allocatable memory |
| `PRE_COMPUTE_IMAGE_MISSING` | `COMPUTING` | `TEE` | The `pre-compute OCI` image could not be found on the worker |

> | `PRE_COMPUTE_TASK_ID_MISSING` | `COMPUTING` | `TEE` | The `IEXEC_TASK_ID` environment variable could not be resolved from the `pre-compute` container |

Reviewer 2: Maybe we coud make something more compact in these blocks, something like:
``IEXEC_XXX` environment variable could not be resolved from `pre-compute`

Author: I agree there is something to think about here.
We could either simplify things as you suggest or be more precise on when the variables exist and their relation to existing deal parameters (e.g. dataset variables should only appear when IS_DATASET_REQUIRED is true).

Maybe cross references could exist between the task-feedback page and the technical reference.

Maybe we should let this as is for the moment and come back to this at a later time. What do you think ?

| `POST_COMPUTE_RESULT_DIGEST_COMPUTATION_FAILED` | `COMPUTING` | | The result digest could not be computed from the `computed.json` file |
| `POST_COMPUTE_OUT_FOLDER_ZIP_FAILED` | `COMPUTING` | | `post-compute` failed to zip the output folder resulting from the computation |
| `POST_COMPUTE_ENCRYPTION_FAILED` | `COMPUTING` | `TEE` | Results encryption failed in `post-compute` container |
| `POST_COMPUTE_RESULT_FILE_NOT_FOUND` | `COMPUTING` | `TEE` | Local result file to upload could not be found |
| `POST_COMPUTE_DROPBOX_UPLOAD_FAILED` | `COMPUTING` | `TEE` | Upload to `DROPBOX` failed |
| `POST_COMPUTE_IPFS_UPLOAD_FAILED` | `COMPUTING` | `TEE` | Upload to `IPFS` failed |
| `POST_COMPUTE_INVALID_TEE_SIGNATURE` | `COMPUTING` | `TEE` | The provided signed TEE challenge is invalid |
| `POST_COMPUTE_SEND_COMPUTED_FILE_FAILED` | `COMPUTING` | | Failed to post `computed.json` to worker |
| `POST_COMPUTE_TIMEOUT` | `COMPUTING` | `TEE` | |
| `POST_COMPUTE_FAILED_UNKOWN_ISSUE` | `COMPUTING` | `TEE` | |
| `OUT_OF_GAS` | `CONTRIBUTING`, `REVEALING` | | The worker needs some ETH, please refill its wallet |

> | `DETERMINISM_HASH_NOT_FOUND` | `CONTRIBUTING`, `REVEALING` | | The `computed.json` file does not contain the hash |

Reviewer 2: ```suggestion
| `DETERMINISM_HASH_NOT_FOUND` | `CONTRIBUTING`, `REVEALING` | | The `computed.json` file does not contain result hash |
```

| `CHAIN_RECEIPT_NOT_VALID` | `CONTRIBUTING`, `REVEALING` | | The transaction failed on the blockchain |
| `ENCLAVE_SIGNATURE_NOT_FOUND` | `CONTRIBUTING` | `TEE` | |
| `CONSENSUS_BLOCK_MISSING` | `REVEALING` | | |
| `BLOCK_NOT_REACHED` | `REVEALING` | | The worker has not reached the consensus block |
| `CANNOT_REVEAL` | `REVEALING` | | One of the mandatory condition was not met. Reveal cannot happen |

> | `RESULT_LINK_MISSING` | `UPLOADING` | | No upload result link was determined for the task |

Reviewer 2: ```suggestion
| `RESULT_LINK_MISSING` | `UPLOADING` | | No upload result link was provided by the worker |
```

| `COMPUTING` | The worker is computing the _task_ |
| `COMPUTED` | The computation is completed |
| `COMPUTE_FAILED` | The computation failed |
| `CONTRIBUTING` | The worker sent the "contribute(..)" transaction (result digest) on chain |
| `CONTRIBUTE_FAILED` | The contribute transaction failed |
| `CONTRIBUTED` | The worker has contributed on chain |
| `REVEALING` | The worker sent the "reveal(..)" transaction (proof that he is the owner of the result digest) |
| `REVEALED` | The worker has revealed the proof on chain |
| `REVEAL_FAILED` | The reveal transaction failed |
| `RESULT_UPLOAD_REQUESTED` | The worker has been requested to upload the result to a remote filesystem |
| `RESULT_UPLOAD_REQUEST_FAILED` | The worker did not accept to be requested to upload the result |
| `RESULT_UPLOADING` | The worker is uploading the result |
| `RESULT_UPLOAD_FAILED` | The upload of the result failed |

> | `RESULT_UPLOADED` | The result has been uploaded to IPFS over the _iExec Result Proxy_ (standard or TEE tasks) or to Dropbox (TEE only), dependending on the _deal_ parameters |

Reviewer 2: Couple occurrences of `_task_` and` _tasks_ ` are missing

Author: Thanks for the feedback, please see 4247621

#### Specific failures for standard tasks

| Replicate failure cause | Replicate status | Description |
| --- | --- | --- |
| `TASK_DESCRIPTION_INVALID` | `STARTING` | The task description contains inconsistencies and cannot be executed |
| `DATASET_FILE_DOWNLOAD_FAILED` | `DATA_DOWNLOADING` | Dataset download failed |
| `DATASET_FILE_BAD_CHECKSUM` | `DATA_DOWNLOADING` | Downloaded dataset checksum does not match on-chain provided checksum |
| `INPUT_FILES_DOWNLOAD_FAILED` | `DATA_DOWNLOADING` | At least one input file could not be downloaded |

#### Specific failures for TEE tasks

| Replicate failure cause | Replicate status | Description |
| --- | --- | --- |
| `TEE_NOT_SUPPORTED` | `STARTING` | The current worker does not support `TEE` tasks. It may not be well configured or not compatible at all |
| `UNKNOWN_SMS` | `STARTING` | SMS URL could not be resolved for this task |
| `GET_TEE_SERVICES_CONFIGURATION_FAILED` | `STARTING` | Failed to fetch TEE task configuration properties from SMS |
| `TEE_PREPARATION_FAILED` | `COMPUTING` | TEE task preparation step could not be completed, task cannot be executed by the worker |

> | `TEE_SESSION_GENERATION_INVALID_AUTHORIZATION` | `COMPUTING` | The worker could not authenticate on the SMS |

Reviewer 2: ```suggestion
| `TEE_SESSION_GENERATION_INVALID_AUTHORIZATION` | `COMPUTING` | The worker could not authenticate to the SMS |
```

| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_EMPTY_PARAMS_UNAUTHORIZED` | `COMPUTING` | The parameters provided by the worker do not allow the `TEE` session generation |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_NO_MATCH_ONCHAIN_TYPE` | `COMPUTING` | The provided task id does not match with a `TEE` task on-chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_GET_CHAIN_TASK_FAILED` | `COMPUTING` | The associated _task_ could not be retrieved on-chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_TASK_NOT_ACTIVE` | `COMPUTING` | The _task_ is not active on chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_GET_CHAIN_DEAL_FAILED` | `COMPUTING` | The associated _deal_ could not be retrieved on-chain |

> | `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_INVALID_SIGNATURE` | `COMPUTING` | The provided authorization was not signed by the correct workerpool |

Reviewer 2: ```suggestion
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_INVALID_SIGNATURE` | `COMPUTING` | The provided authorization has not been signed by the correct workerpool owner |
```

| `TEE_SESSION_GENERATION_PRE_COMPUTE_GET_DATASET_SECRET_FAILED` | `COMPUTING` | Dataset secret could not be fetched from the SMS database |

> | `TEE_SESSION_GENERATION_APP_COMPUTE_NO_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be read from _task_ description |

Reviewer 2: ```suggestion
| `TEE_SESSION_GENERATION_APP_COMPUTE_NO_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be read from on-chain _app_ description |
```

| `TEE_SESSION_GENERATION_APP_COMPUTE_INVALID_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be validated |

> | `TEE_SESSION_GENERATION_POST_COMPUTE_GET_ENCRYPTION_TOKENS_FAILED_EMPTY_BENEFICIARY_KEY` | `COMPUTING` | Beneficiary results encryption key could not be fetched from the SMS database |

Reviewer 2: ```suggestion
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_ENCRYPTION_TOKENS_FAILED_EMPTY_BENEFICIARY_KEY` | `COMPUTING` | Beneficiary result encryption key could not be fetched from the SMS database |
```

| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_STORAGE_TOKENS_FAILED` | `COMPUTING` | Results storage (IPFS or Dropbox) token could not be fetched from the SMS database |

> | `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_WORKER_ADDRESS` | `COMPUTING` | The worker address is missing for the enclave signature computation |

Reviewer 2: ```suggestion
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_WORKER_ADDRESS` | `COMPUTING` | The worker address is missing to compute the enclave signature |
```

| `POST_COMPUTE_TEE_CHALLENGE_PRIVATE_KEY_MISSING` | `COMPUTING` | The `RESULT_SIGN_TEE_CHALLENGE_PRIVATE_KEY` environment variable could not be resolved from the `post-compute` container |
| `POST_COMPUTE_ENCRYPTION_PUBLIC_KEY_MISSING` | `COMPUTING` | The `RESULT_ENCRYPTION_PUBLIC_KEY` environment variable could not be resolved from the `post-compute` container |
| `POST_COMPUTE_STORAGE_TOKEN_MISSING` | `COMPUTING` | The `RESULT_STORAGE_TOKEN` environment variable could not be resolved from the `post-compute` container |
| `POST_COMPUTE_ENCRYPTION_FAILED` | `COMPUTING` | Results encryption failed in `post-compute` container |
| `POST_COMPUTE_RESULT_FILE_NOT_FOUND` | `COMPUTING` | Local result file to upload could not be found |
| `POST_COMPUTE_DROPBOX_UPLOAD_FAILED` | `COMPUTING` | Upload to `DROPBOX` failed |
| `POST_COMPUTE_IPFS_UPLOAD_FAILED` | `COMPUTING` | Upload to `IPFS` failed |
| `POST_COMPUTE_INVALID_TEE_SIGNATURE` | `COMPUTING` | The provided signed TEE challenge is invalid |
| `POST_COMPUTE_TIMEOUT` | `COMPUTING` | The `post-compute` did not complete on time |
| `POST_COMPUTE_FAILED_UNKOWN_ISSUE` | `COMPUTING` | The `post-compute` failed for an unknown cause |

> | `ENCLAVE_SIGNATURE_NOT_FOUND` | `CONTRIBUTING` | The enclave signature could not be verified on the results. The latter may have been tampered with and will not be contributed |

Reviewer 2: ```suggestion
| `ENCLAVE_SIGNATURE_NOT_FOUND` | `CONTRIBUTING` | The enclave signature could not be found for the result. The worker will not contribute |
```

| `CONTRIBUTION_ALREADY_SET` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The worker has already contributed |
| `WORKERPOOL_AUTHORIZATION_NOT_FOUND` | `STARTING`, `APP_DOWNLOADING`, `DATA_DOWNLOADING`, `COMPUTING`, `CONTRIBUTING` | | The authorization to contribute to the task is missing |
| `TASK_DESCRIPTION_INVALID` | `STARTING` | `STD` | The task description contains inconsistencies and cannot be executed |

> | `TEE_NOT_SUPPORTED` | `STARTING` | `TEE` | The current worker is not well configured to execute TEE tasks (or not compatible at all) |

Reviewer: What about the following?
```suggestion
| `TEE_NOT_SUPPORTED` | `STARTING` | `TEE` | The current worker does not support TEE tasks. It may not be well configured or not compatible at all |
```

Author: What do you think of 26a5560 ?

| `UNKNOWN_SMS` | `STARTING` | SMS URL could not be resolved for this task |
| `GET_TEE_SERVICES_CONFIGURATION_FAILED` | `STARTING` | Failed to fetch TEE task configuration properties from SMS |
| `TEE_PREPARATION_FAILED` | `COMPUTING` | TEE task preparation step could not be completed, task cannot be executed by the worker |
| `TEE_SESSION_GENERATION_INVALID_AUTHORIZATION` | `COMPUTING` | The worker could not authenticate on the SMS |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_EMPTY_PARAMS_UNAUTHORIZED` | `COMPUTING` | The parameters provided by the worker do not allow the `TEE` session generation |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_NO_MATCH_ONCHAIN_TYPE` | `COMPUTING` | The provided task id does not match with a `TEE` task on-chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_GET_CHAIN_TASK_FAILED` | `COMPUTING` | The associated _task_ could not be retrieved on-chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_TASK_NOT_ACTIVE` | `COMPUTING` | The _task_ is not active on chain |
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_GET_CHAIN_DEAL_FAILED` | `COMPUTING` | The associated _deal_ could not be retrieved on-chain |

> | `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_INVALID_SIGNATURE` | `COMPUTING` | The provided authorization was not signed by the correct workerpool |

Reviewer: ```suggestion
| `TEE_SESSION_GENERATION_EXECUTION_NOT_AUTHORIZED_INVALID_SIGNATURE` | `COMPUTING` | The provided authorization has not been signed by the correct workerpool |
```

Author: fixed in 7dbba92

| `TEE_SESSION_GENERATION_PRE_COMPUTE_GET_DATASET_SECRET_FAILED` | `COMPUTING` | Dataset secret could not be fetched from the SMS database |
| `TEE_SESSION_GENERATION_APP_COMPUTE_NO_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be read from _task_ description |

> | `TEE_SESSION_GENERATION_APP_COMPUTE_INVALID_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be validated |

Reviewer: Do we have more info on why it has not been validated? Which values may cause that error to happen?

Author: What do you think of this ?
| `TEE_SESSION_GENERATION_APP_COMPUTE_INVALID_ENCLAVE_CONFIG` | `COMPUTING` | Application enclave configuration could not be validated. At least one item between the entrypoint, heap size or fingerprint of the _app_ enclave description is not correct |

Reviewer: > At least one item among the entrypoint, the heap size or the fingerprint of the app enclave configuration is not correct

| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_STORAGE_TOKENS_FAILED` | `COMPUTING` | Results storage (IPFS or Dropbox) token could not be fetched from the SMS database |
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_WORKER_ADDRESS` | `COMPUTING` | The worker address is missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_PUBLIC_ENCLAVE_CHALLENGE` | `COMPUTING` | The enclave challenge is missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CHALLENGE` | `COMPUTING` | The `TEE` challenge is missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CREDENTIALS` | `COMPUTING` | The `TEE` credentials are missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_SECURE_SESSION_STORAGE_CALL_FAILED` | `COMPUTING` | The `TEE` session could not be posted on the secure session storage |
| `TEE_SESSION_GENERATION_SECURE_SESSION_GENERATION_FAILED` | `COMPUTING` | |

> | `TEE_SESSION_GENERATION_SECURE_SESSION_NO_TEE_PROVIDER` | `COMPUTING` | No `TEE` provider was found on the _task_ description |

Reviewer: ```suggestion
| `TEE_SESSION_GENERATION_SECURE_SESSION_NO_TEE_PROVIDER` | `COMPUTING` | No `TEE` provider has found been on the _task_ description |
```

Author: Thanks, please see c68d87a

| `TEE_SESSION_GENERATION_GET_TASK_DESCRIPTION_FAILED` | `COMPUTING` | The _task_ description could not be read on-chain |
| `TEE_SESSION_GENERATION_NO_SESSION_REQUEST` | `COMPUTING` | The `TEE` session request was empty |

> | `TEE_SESSION_GENERATION_NO_TASK_DESCRIPTION` | `COMPUTING` | The _task_ description was not found in the `TEE` session request |

Reviewer: ```suggestion
| `TEE_SESSION_GENERATION_NO_TASK_DESCRIPTION` | `COMPUTING` | The _task_ description has not been found in the `TEE` session request |
```

Author: Thanks, please see c68d87a

| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CHALLENGE` | `COMPUTING` | The `TEE` challenge is missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_POST_COMPUTE_GET_SIGNATURE_TOKENS_FAILED_EMPTY_TEE_CREDENTIALS` | `COMPUTING` | The `TEE` credentials are missing for the enclave signature computation |
| `TEE_SESSION_GENERATION_SECURE_SESSION_STORAGE_CALL_FAILED` | `COMPUTING` | The `TEE` session could not be posted on the secure session storage |

> | `TEE_SESSION_GENERATION_SECURE_SESSION_GENERATION_FAILED` | `COMPUTING` | The `TEE` session could not be generate for an unknown reason |

Reviewer: ```suggestion
| `TEE_SESSION_GENERATION_SECURE_SESSION_GENERATION_FAILED` | `COMPUTING` | The `TEE` session could not be generated but the SMS can't provide the reason |
```

| `TEE_SESSION_GENERATION_SECURE_SESSION_NO_TEE_PROVIDER` | `COMPUTING` | No `TEE` provider was found on the _task_ description |
| `TEE_SESSION_GENERATION_GET_TASK_DESCRIPTION_FAILED` | `COMPUTING` | The _task_ description could not be read on-chain |
| `TEE_SESSION_GENERATION_NO_SESSION_REQUEST` | `COMPUTING` | The `TEE` session request was empty |
| `TEE_SESSION_GENERATION_NO_TASK_DESCRIPTION` | `COMPUTING` | The _task_ description was not found in the `TEE` session request |

> | `TEE_SESSION_GENERATION_UNKNOWN_ISSUE` | `COMPUTING` | The `TEE` session could not be generated for an unknown reason |

Reviewer: ```suggestion
| `TEE_SESSION_GENERATION_UNKNOWN_ISSUE` | `COMPUTING` | The `TEE` session could not be generated but the SMS hasn't provided any reason |
```

| `PRE_COMPUTE_DATASET_FILENAME_MISSING` | `COMPUTING` | The `IEXEC_DATASET_FILENAME` environment variable could not be resolved from the `pre-compute` container |
| `PRE_COMPUTE_INPUT_FILES_NUMBER_MISSING` | `COMPUTING` | The `IEXEC_INPUT_FILES_NUMBER` environment variable could not be resolved from the `pre-compute` container |
| `PRE_COMPUTE_AT_LEAST_ONE_INPUT_FILE_URL_MISSING` | `COMPUTING` | At least one environment variable starting with `IEXEC_INPUT_FILE_URL_` prefix could not be resolve from the `pre-compute` container |
| `PRE_COMPUTE_OUTPUT_FOLDER_NOT_FOUND` | `COMPUTING` | The folder to store the dataset or input files could not be found |
| `PRE_COMPUTE_DATASET_DOWNLOAD_FAILED` | `COMPUTING` | The `pre-compute` failed to download the dataset |
| `PRE_COMPUTE_INVALID_DATASET_CHECKSUM` | `COMPUTING` | The downloaded dataset checksum does not match the expected one |
| `PRE_COMPUTE_DATASET_DECRYPTION_FAILED` | `COMPUTING` | The downloaded dataset decryption failed |

> | `PRE_COMPUTE_SAVING_PLAIN_DATASET_FAILED` | `COMPUTING` | The decrypted dataset could not be saved |

Reviewer: I would insist on the fact that the plain dataset is saved in the enclave:
```suggestion
| `PRE_COMPUTE_SAVING_PLAIN_DATASET_FAILED` | `COMPUTING` | The decrypted dataset could not be saved in the enclave |
```

| `POST_COMPUTE_TIMEOUT` | `COMPUTING` | `TEE` | |
| `POST_COMPUTE_FAILED_UNKOWN_ISSUE` | `COMPUTING` | `TEE` | |
| `OUT_OF_GAS` | `CONTRIBUTING`, `REVEALING` | | The worker needs some ETH, please refill its wallet |

> | `DETERMINISM_HASH_NOT_FOUND` | `CONTRIBUTING`, `REVEALING` | | The `computed.json` file does not contain result hash |

Author: What do you think of 26a5560 ?

| `CHAIN_RECEIPT_NOT_VALID` | `CONTRIBUTING`, `REVEALING` | | The transaction failed on the blockchain |
| `ENCLAVE_SIGNATURE_NOT_FOUND` | `CONTRIBUTING` | `TEE` | |
| `CONSENSUS_BLOCK_MISSING` | `REVEALING` | | |
| `BLOCK_NOT_REACHED` | `REVEALING` | | The worker has not reached the consensus block |
| `CANNOT_REVEAL` | `REVEALING` | | One of the mandatory condition was not met. Reveal cannot happen |

> | `RESULT_LINK_MISSING` | `UPLOADING` | | No upload result link was provided by the worker |

Author: please see da373df
